juju/client/proxy/proxy.py

```diff
@@ -0,0 +1,17 @@
+from abc import ABC, abstractmethod
```

Comment:
You generally don't use ABC's in Python (duck typing FTW). You'd just define and document the class, and people can subclass it or define a class that quacks like it if they want. I think you should just drop this abstract class (the rest of the project doesn't use them).